mirrord-agent/src/main.rs

```diff
@@ -160,6 +231,33 @@ async fn handle_peer_messages(
                 .send((peer_message.peer_id, file_request))
                 .await?;
         }
+        ClientMessage::OverrideEnvVarsRequest(OverrideEnvVarsRequest { filter_env_vars }) => {
+            debug!(
+                "ClientMessage::OverrideEnvVarsRequest peer id {:?} requesting {:?}",
+                peer_message.peer_id, filter_env_vars
+            );
+
+            let container_runtime = container_runtime
```

Comment:
maybe start the abstraction with this PR and create a `EnvVarHandler` or smth similar that holds all the needed stuff (container id, runtime, etc?) and just handles the message directly (to reduce logic congestion here :P)
not top priority, but it's a refactor we would like to have soon anyway

Reply:
As this PR took a bit too long, I'll leave this up as a TODO for now.